arex-web-core/src/main/java/com/arextest/web/core/business/util/SchemaUtils.java

```diff
@@ -0,0 +1,135 @@
+        if (value == null) {
+            return;
+        }
+        if (contract == null) {
+            contract = new HashMap<>();
+        }
+        Object contractItem = contract.get(key);
+        if (value instanceof Map<?, ?>) {
+            Map<String, Object> mapContract;
+            if (contractItem instanceof Map<?, ?>) {
+                mapContract = (Map<String, Object>) contractItem;
+            } else {
+                mapContract = new HashMap<>();
+            }
+            mergeMap(mapContract, (Map<String, Object>) value);
+            contract.put(key, mapContract);
+        } else if (value instanceof List) {
+            List<Object> listContract;
+            if (contractItem instanceof List) {
+                listContract = (List<Object>) contractItem;
+            } else {
+                listContract = new ArrayList<>();
+            }
+            mergeList(listContract, (List<Object>) value);
+            contract.put(key, listContract);
+        } else {
+            if (contract.get(key) == null) {
+                contract.put(key, handlePrimaryItem(value));
+            }
+        }
+    }
+
+    private static void mergeList(List<Object> contract, List<Object> model) {
+        Object contractItem = null;
+        if (!contract.isEmpty()) {
+            contractItem = contract.remove(0);
+        }
+
+        for (Object item : model) {
+            if (item instanceof Map<?, ?>) {
+                Map<String, Object> mapContract;
+                if (contractItem instanceof Map<?, ?>) {
+                    mapContract = (Map<String, Object>) contractItem;
+                } else {
+                    mapContract = new HashMap<>();
+                }
+                mergeMap(mapContract, (Map<String, Object>) item);
+            } else if (item instanceof List) {
+                List<Object> listContract;
+                if (contractItem instanceof List) {
+                    listContract = (List<Object>) contractItem;
+                } else {
+                    listContract = new ArrayList<>();
+                }
+                mergeList(listContract, (List<Object>) item);
+            } else {
+                if (contractItem == null) {
+                    contractItem = handlePrimaryItem(item);
+                }
+            }
+        }
+        if (contractItem != null) {
+            contract.add(contractItem);
+        }
+    }
+
+    private static Object handlePrimaryItem(Object item) {
+        if (item instanceof Integer || item instanceof Long) {
+            return DEFAULT_INT;
+        }
+        if (item instanceof Double || item instanceof Float) {
+            return DEFAULT_DOUBLE;
+        }
+        if (item instanceof Boolean) {
+            return Boolean.TRUE;
+        }
+        if (item instanceof String) {
+            return DEFAULT_STR;
+        }
+        if (item instanceof char[]) {
```

Comment:
after jackson deserialization, there are leaf nodes as char[]?

Reply:
After test, there won't be char[]